src/include/daos/mem.h

```diff
@@ -862,4 +863,217 @@ struct umem_action {
 	};
 };
 
+#define UMEM_CACHE_PAGE_SZ_SHIFT  24 /* 16MB */
+#define UMEM_CACHE_PAGE_SZ        (1 << UMEM_CACHE_PAGE_SZ_SHIFT)
+#define UMEM_CACHE_PAGE_SZ_MASK   (UMEM_CACHE_PAGE_SZ - 1)
+
+#define UMEM_CACHE_CHUNK_SZ_SHIFT 14 /* 16KB */
+#define UMEM_CACHE_CHUNK_SZ       (1 << UMEM_CACHE_CHUNK_SZ_SHIFT)
+#define UMEM_CACHE_CHUNK_SZ_MASK  (UMEM_CACHE_CHUNK_SZ - 1)
+
+#define UMEM_CACHE_BMAP_SZ        (1 << (UMEM_CACHE_PAGE_SZ_SHIFT - UMEM_CACHE_CHUNK_SZ_SHIFT - 6))
+
+/** 16 MB page */
+struct umem_page {
+	/** page ID */
+	unsigned int		 pg_id;
+	/** refcount */
+	int			 pg_ref;
+	uint64_t                 pg_dirty : 1, /** Page is dirty */
```

Comment:
The dirty bit looks redundant to me, we could see if the page is 'dirty' by comparing the 'pg_last_inflight' with global 'last checkpointed ID', right?

Reply:
Removed.